app/models/collection_participant.rb

```diff
@@ -57,4 +59,11 @@ def user_allowed_to_promote?(user, role)
     (role == MEMBER || role == NONE) ? self.collection.user_is_maintainer?(user) : self.collection.user_is_owner?(user)
   end
 
+  def reindex_collection
```

Comment:
It looks like in some places, the method in charge of the reindexing is called `reindex_collection`, but in others, it's `update_collection_index` (or similar). I don't immediately see a reason for this distinction... is there one? If not, could we call it `update_collection_index` everywhere?

Reply:
I don't think there's a reason, we were just inconsistent (CommonTagging adds `update_search` into the mix). But using `update_*_index` for the new callbacks sounds good.